### ISSUE TITLE
Warning: Cannot assign to read only property 'subarray' of main.js

### ISSUE THREAD

**Issue by Phinome** [gruntjs/grunt-contrib-uglify#315]
I recently updated to node 0.12.1, npm 2.7.3 and right now I’m getting the following error with grunt-contrib-uglify version 0.8.0:

Running "uglify:min" (uglify) task
[Warnings: Cannot assign to read only property 'subarray' of main.js' content]
>> Uglifying source static/js/main.js failed.
Warning: Uglification failed.
Cannot assign to read only property 'subarray' of main.js' content.
 Use --force to continue.

Aborted due to warnings.
Happens with any .js file, so it seems to be a bug either with nodejs v0.12.1 or with this uglify task. Any ideas?

here is my gruntfile.js configuration:
```
uglify: {
      options: {
        banner: '<%= banner %>'
      },
      js: {
        src : '<%= concat.js.dest %>',
        dest : 'release/<%= pkg.name %>.min.js'
      }
    }
```

**Comment by jrhite** [comment]
There was a recent upgrade of the underlying uglifyjs tool. The version of
uglify that grunt-contrib-uglify uses is ^2.4.0 which means it's picking up
the latest uglifyjs 2.4.19 changes.

I recommend this PR
<https://github.com/gruntjs/grunt-contrib-uglify/pull/314> from @alonisser gets
approved/submitted asap so that the uglifyjs tools gets pinned to 2.4.17
for now.

Also can you please repro this bug with a small sample of code (no grunt
included -- just use the uglifyjs cli interface directly) and submit a bug
here: https://github.com/mishoo/UglifyJS2

**Comment by Phinome** [comment]
@jrhite that's ok , when i use the uglifyjs cli interface directly. but i do not understand these meanings and next what i should do ?

>
I recommend this PR
<https://github.com/gruntjs/grunt-contrib-uglify/pull/314> from @alonisser gets
approved/submitted asap so that the uglifyjs tools gets pinned to 2.4.17
for now.

because i see this PR <https://github.com/gruntjs/grunt-contrib-uglify/pull/314> displayed 
>
This branch has failed checks, but can be merged.


That's all , Thanks.

**Comment by jrhite** [comment]
The tests you see failing, have been failing for about 17 days now and are likely unrelated to the problems you have recently starting seeing.

A bug may have been created in the way the grunt-contrib-uglify interacts with the underlying uglifyjs api. This bug was likely created due to the new uglify release and grunt-contrib-uglify will need to be updated (I have a PR to update it). For now, to get around this we can 'FIX' grunt-contrib-uglify to version 2.4.17 which doesn't have the potential grunt-contrib-uglify/uglifyjs tool conflicts.

Currently however grunt-contrib-uglify uses version ^2.4.0 which means it picks up the latest 2.x.x version, which is currently 2.4.19. To avoid this we remove the ^ and specify a specific version, 2.4.17 until the bug(s) in 2.4.19 are worked out. That's what @alonisser 's PR is about.

Does that make sense?

Cheers.

**Comment by Phinome** [comment]
Thanks. I see.
